packages/test/test-end-to-end-tests/src/test/treeDataObject.spec.ts

```diff
@@ -0,0 +1,97 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "assert";
+
+import { describeCompat } from "@fluid-private/test-version-utils";
+import {
+	ContainerRuntimeFactoryWithDefaultDataStore,
+	PureDataObjectFactory,
+	TreeDataObject,
+} from "@fluidframework/aqueduct/internal";
+import type { IContainer } from "@fluidframework/container-definitions/internal";
+import type { IContainerRuntimeOptions } from "@fluidframework/container-runtime/internal";
+import {
+	createContainerRuntimeFactoryWithDefaultDataStore,
+	getContainerEntryPointBackCompat,
+	type ITestObjectProvider,
+} from "@fluidframework/test-utils/internal";
+import {
+	SchemaFactory,
+	SharedTree,
+	TreeViewConfiguration,
+	type ITree,
+	type TreeView,
+} from "@fluidframework/tree/internal";
+
+const schemaFactory = new SchemaFactory("test");
+class TestSchema extends schemaFactory.object("TestSchema", {
+	foo: [schemaFactory.string, schemaFactory.number],
+}) {}
+
+const treeViewConfig = new TreeViewConfiguration({ schema: TestSchema });
+
+class TestTreeDataObject extends TreeDataObject<TreeView<typeof TestSchema>> {
+	public override generateView(tree: ITree): TreeView<typeof TestSchema> {
+		return tree.viewWith(treeViewConfig);
+	}
+
+	public override async initializingFirstTime(): Promise<void> {
+		assert(this.treeView.compatibility.canInitialize);
+		this.treeView.initialize({ foo: "Hello world" });
+	}
+
+	public static readonly type = "TestTreeDataObject";
+
+	public static getFactory(): PureDataObjectFactory<TestTreeDataObject> {
+		return TestTreeDataObject.factory;
+	}
+
+	private static readonly factory = new PureDataObjectFactory(
+		TestTreeDataObject.type,
+		TestTreeDataObject,
+		[SharedTree.getFactory()],
+		{},
+	);
+}
+
+// Note: ideally these tests would live directly in the `aqueduct` package,
+// but much of the test infrastructure used below is not reachable from that package.
+describeCompat("TreeDataObject", "NoCompat", (getTestObjectProvider) => {
+	// Runtime ID compression is required to use SharedTree.
+	const runtimeOptions: IContainerRuntimeOptions = {
+		enableRuntimeIdCompressor: "on",
+	};
+
+	const runtimeFactory = createContainerRuntimeFactoryWithDefaultDataStore(
+		ContainerRuntimeFactoryWithDefaultDataStore,
+		{
+			defaultFactory: TestTreeDataObject.getFactory(),
+			registryEntries: [[TestTreeDataObject.type, TestTreeDataObject.getFactory()]],
+			runtimeOptions,
+		},
+	);
+
+	let provider: ITestObjectProvider;
+	let container: IContainer;
+	beforeEach(async () => {
+		provider = getTestObjectProvider();
+		container = await provider.createContainer(runtimeFactory);
+	});
```

Comment:
I'm not a fan of the have mocha hooks modify mutable variables outside of tests pattern.

Is there a reason you can't just duplicate these variables in each test? Its actually less code, more self contained, easier to modify and easier to follow and read in my opinion.

More comments on this subject are in https://github.com/microsoft/FluidFramework/pull/24016#discussion_r1992336075

Reply:
Fair. It's the pattern used elsewhere, so I just adopted it here. I tend to agree with you, though I think there is also a fair argument for prioritizing consistency.

Reply:
Updated.